File: frontend/scripts/test-route-transition.mjs
import { chromium } from 'playwright'
import { mkdirSync } from 'fs'

const out = '.screenshots'
mkdirSync(out, { recursive: true })

function curtainDebug(page) {
  return page.evaluate(() => {
    const root = document.querySelector('.route-curtain')
    const panel = document.querySelector('.route-curtain__panel')
    if (!root || !panel) return { missing: true }
    return {
      state: root.dataset.state,
      rootOpacity: getComputedStyle(root).opacity,
      pointerEvents: getComputedStyle(root).pointerEvents,
      panelTransform: panel.style.transform || getComputedStyle(panel).transform,
      path: location.pathname,
    }
  })
}

const browser = await chromium.launch({ headless: true })
const page = await browser.newPage({ viewport: { width: 1440, height: 900 } })

await page.addInitScript(() => {
  sessionStorage.setItem('grunray_splash_woniu_v1', '1')
})

const base = process.env.PREVIEW_URL || 'http://localhost:4173'
await page.goto(`${base}/`, { waitUntil: 'networkidle' })
await page.waitForTimeout(1500)
await page.screenshot({ path: `${out}/01-home.png`, fullPage: true })
console.log('home', JSON.stringify(await curtainDebug(page), null, 2))

// 博客在「创作」下拉中
await page.locator('.nav-group').first().hover()
await page.waitForTimeout(300)
await page.locator('.dropdown-item[href="/blog"]').click()
await page.waitForTimeout(350)
await page.screenshot({ path: `${out}/02-blog-mid.png`, fullPage: true })
console.log('mid', JSON.stringify(await curtainDebug(page), null, 2))

await page.waitForTimeout(900)
await page.screenshot({ path: `${out}/03-blog-after.png`, fullPage: true })
console.log('after', JSON.stringify(await curtainDebug(page), null, 2))

await page.locator('nav a.link[href="/"]').click()
await page.waitForTimeout(350)
await page.screenshot({ path: `${out}/04-home-mid.png`, fullPage: true })
console.log('back-mid', JSON.stringify(await curtainDebug(page), null, 2))

await page.waitForTimeout(900)
await page.screenshot({ path: `${out}/05-home-final.png`, fullPage: true })
console.log('back-final', JSON.stringify(await curtainDebug(page), null, 2))

await browser.close()

File: frontend/scripts/test-route-transition.py
from pathlib import Path

from playwright.sync_api import sync_playwright

OUT = Path(__file__).resolve().parent.parent / ".screenshots"
OUT.mkdir(exist_ok=True)


def curtain_debug(page):
    return page.evaluate(
        """() => {
          const root = document.querySelector('.route-curtain');
          if (!root) return { missing: true };
          const panels = [...document.querySelectorAll('.route-curtain__panel')].map((el) => ({
            inline: el.style.transform,
            computed: getComputedStyle(el).transform,
          }));
          return {
            state: root.dataset.state,
            pointerEvents: getComputedStyle(root).pointerEvents,
            panels,
          };
        }"""
    )


with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page(viewport={"width": 1440, "height": 900})
    page.goto("http://localhost:5173/", wait_until="networkidle")
    page.wait_for_timeout(2500)
    page.screenshot(path=str(OUT / "01-home.png"), full_page=True)

    page.locator('nav a[href="/blog"]').first.click()
    page.wait_for_timeout(400)
    page.screenshot(path=str(OUT / "02-blog-mid.png"), full_page=True)
    print("mid:", curtain_debug(page))

    page.wait_for_timeout(1200)
    page.screenshot(path=str(OUT / "03-blog-after.png"), full_page=True)
    print("after:", curtain_debug(page))
    print("url:", page.url)

    page.locator('nav a[href="/"]').first.click()
    page.wait_for_timeout(1500)
    page.screenshot(path=str(OUT / "04-back-home.png"), full_page=True)
    print("back:", curtain_debug(page))

    browser.close()
